components/smalltalk/squeak5/Makefile

```diff
@@ -0,0 +1,95 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"). You may
+# only use this file in accordance with the terms of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Package Makefile for the "OpenSmalltalk Cog Spur" Squeak Smalltalk V5 system
+# See http://squeak.org and http://opensmalltalk.org
+#
+
+BUILD_BITS=64
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		squeak5
+COMPONENT_VERSION=	5.0.6
+COMPONENT_SUMMARY=	The Squeak V5 Smalltalk Virtual Machine
+COMPONENT_PROJECT_URL=	http://www.squeak.org
+COMPONENT_FMRI=		runtime/squeak5
+COMPONENT_CLASSIFICATION=	Development/Other Languages
+COMPONENT_LICENSE=	LICENSE
+COMPONENT_SRC=		opensmalltalk-vm
+
+GIT=			git
+GIT_REPO=		https://github.com/cstes/opensmalltalk-vm.git
```

Comment:
Can you add --depth=1 for cloning the repository? Cloning the whole history is taking a lot of time and space.

Reply:
I can do that.   I think working with a .tar.gz tarball (COMPONENT_ARCHIVE_URL ) will in the end be easier than working with "git" (in fact git offers the feature with 'releases'  and then downloading it from codeload.github.com) .   But as I wrote before currently I have to build from the git clone and cannot build from an exported .tar.gz which is due to the upstream opensmalltalk-vm build scripts.   I'll test the --depth=1 tomorrow.

Reply:
Yes, working with COMPONENT_ARCHIVE_URL is the preferred way. We have a files server where we put the files the build server fetches so we have an alternative path if the original is not accessible.

Reply:
I have checked the Github project pages. They have releases with source files, eg. https://github.com/OpenSmalltalk/opensmalltalk-vm/releases with https://github.com/OpenSmalltalk/opensmalltalk-vm/archive/202003021730.tar.gz
Maybe you can use this? Taking sources from a release and not the latest revision may also be wiser and less error prone.

Reply:
I started by doing that, before I used the "git clone".   So I tested it but it does not seem to work.   The problem is - it seems - that scripts/updateSCCSVersions must be ran before configure (configure checks and aborts if it did not run).   Now that script seems to assume that it can issue git commands that fail if it is running on the unversion-end tree.   But I agree that this is something that can be fixed (and should not be so difficult).  And it will be a big win for this package Makefile.   That's true.   I tried working with COMPONENT_ARCHIVE_URL and it first the oi-userland much better, that's true.  I will try to do that if I can convince the "upstream" opensmalltalk of this.    The releases they have on their github.com page have this problem for me ...   I don't know how the Microsoft VisualC guys (who do not use UNIX "configure") use those releases but for the world of UNIX "configure" it does not currently work - unfortunately.    I agree that they should fix that.

Reply:
I tested : git clone --depth 1 --branch sun
That works but it is not much faster and it is not really checking out the tag sun-v5.0.6
Maybe I'm imaging things but what seems to work and what seems faster is
git clone --depth 1 --branch sun-v5.0.6  (which is not the branch but the tag)
So I'll use that.

Reply:
No sorry it does not work;   I think I'll leave the "git clone" as it is right now.   I agree with you that I should work on making COMPONENT_ARCHIVE_URL work .  That fits the oi-userland framework best.   I think it can be done ...   it's currently not how their opensmalltalk-vm "configure" works, but should be possible.

Reply:
I just commited the original code which was using COMPONENT_ARCHIVE_URL.   I think we both agree it 's best.   It's much faster than GIT and using git does not make sense for the oi-userland download. You'll notice that it currently breaks the build, but it should be possible to fix the opensmalltalk-vm configure script, so I know it's currently broken but I'm working on the fix for that.

Reply:
perfect

Reply:
The build should be much faster now because it does not "git clone" any longer,  it works for me ...  The solution to use "patches" is more work for me for subsequent updates (e.g. for version 5.0.7) but the advantage is that it is much faster for rebuilds for userland, I think .   Also, the opensmalltalk project may be interested and they may have a solution on the long term that avoids this problem.   For now, I currently work with 'patches' which is a solution to search a "balance" between the requirements of the userland framework and the requirements of the opensmalltalk project.

Reply:
Yes, this sounds by far better. BTW: You should change to either IRC #oi-dev or mail oi-dev@openindiana.org for communication. IRC is logged by echelog.com so you can read older messages. Albeit the service will close soon and there are discussions on #illumos to provide our own service for it.

Reply:
This seems to work a lot better. Now that you have a fixed version you need to update your sample-manifest.p5m because it has changed.